docs/scripts/api/buildComponentsDocumentation.ts

```diff
@@ -145,7 +146,12 @@ function extractSlots(options: {
       return;
     }
 
-    slots[name] = {
+    // Workaround to generate correct (camelCase) keys for slots in v6 `API Reference` documentation
+    // TODO v7: Remove this condition when `Grid(Pro|Premium)SlotsComponent` type is refactored to have `camelCase` names
+    // Shifting to `slots` prop instead of `components` prop strips off the `default` property due to deduced type `UncapitalizedGridSlotsComponent`
```

Comment:
Thanks for the suggestion, I also considered doing it this way.

There could be two possibilities in this regard:
1. We convert object keys to camelCase for both `GridSlotsComponent` type and `DATA_GRID_DEFAULT_SLOTS_COMPONENTS` (and similar variables/types for Pro and Premium), then to avoid making it a breaking change we'll either have to:
   1. Maintain two (x 3) lists `DATA_GRID_DEFAULT_SLOTS_COMPONENTS: CapitalizedGridSlotsComponent (deduced type)` and `DATA_GRID_DEFAULT_SLOTS_COMPONENTS_UNCAPITALIZED: GridSlotsComponent`
   2. Or somehow apply a util like [uncapitalizeObjectKeys](https://github.com/mui/mui-x/blob/7a4a02640367c63cacfa1662e6236f32b4d74f2c/packages/grid/x-data-grid/src/internals/slotsMigration.ts#L26) (for actually capitalizing the keys) before exporting the `DATA_GRID_DEFAULT_SLOTS_COMPONENTS`
2. We only convert the type and not the `DATA_GRID_DEFAULT_SLOTS_COMPONENTS` (and similar) variables. It also could have these caveats:
   1. Assigning a different type to an existing exported variable (DATA_GRID_DEFAULT_SLOTS_COMPONENTS: CapitalizedGridSlotsComponent) may be considered as a breaking change as users will have to do changes if they are reusing both `DATA_GRID_DEFAULT_SLOTS_COMPONENTS` and `GridSlotsComponent`
   2. Still half of the conversion will be done (only type) and there might have more hassle to figure out connection between the variables and types during v7 preparation.

So I thought the reasonable middle-ground could be to keep the existing variables `GridSlotsComponent` type and `DATA_GRID_DEFAULT_SLOTS_COMPONENTS` as it is, use deduced type and document the needed changes in v7 in an issue (which I will take care of). And I hope the changes won't be that complex and could even be done via a codemod. 🙈 Let me know if you disagree though.

P.S: If I'm not mistaken the same path is taken by `pickers` too.

Reply:
Thanks for the extensive explanation, sounds good to me 👍